index.js

```diff
@@ -25,15 +25,22 @@ const evaluateTemplateLiteral = expression => {
 
 // Uses backtracking DFS to traverse object graph
 const sro = object => {
+  const parsedValues = {};
   const checkedReferences = [];
 
   const parseValue= (value) => {
     if (typeof value !== 'string') {
       return value;
     }
     // lookup any `this` values
-    const expression = value.replace(/this\.[A-Za-z0-9_.]+/g, thisMatch => {
-      return JSON.stringify(lookup(thisMatch.slice(5)));
+    const expression = value.replace(/this\.[A-Za-z0-9_.-]+/g, thisMatch => {
+      const key = thisMatch.slice(5)
+
+      if (!Object.prototype.hasOwnProperty.call(parsedValues, key)) {
```

Comment:
I think `if (!parsedValues.hasOwnProperty(key))` is simpler and clearer

Reply:
Hi @alex-e-leon.
Thank you for pointing that out.
Because [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins) occur in eslint, I always write like this.

Reply:
Good point! This repo could use some linting : )